pkg/scheduler/framework/runtime/framework.go

```diff
@@ -928,8 +935,15 @@ func (f *frameworkImpl) RunScorePlugins(ctx context.Context, state *framework.Cy
 		metrics.FrameworkExtensionPointDuration.WithLabelValues(score, status.Code().String(), f.profileName).Observe(metrics.SinceInSeconds(startTime))
 	}()
 	allNodePluginScores := make([]framework.NodePluginScores, len(nodes))
-	pluginToNodeScores := make(map[string]framework.NodeScoreList, len(f.scorePlugins))
+	var plugins []framework.ScorePlugin
 	for _, pl := range f.scorePlugins {
+		if state.SkipScorePlugins.Has(pl.Name()) {
```

Comment:
Regardless of whether my concern is true or not, it's better to add the test somewhere to confirm "if all PreScore plugins return skip, one Node is selected correctly from the `schedulePod` func."

Reply:
> Regardless of whether my concern is true or not, it's better to add the test somewhere to confirm "if all PreScore plugins return skip, one Node is selected correctly from the `schedulePod` func."

I'm working on it 🚀

Reply:
> I'm guessing empty list is returned from `RunScorePlugins` and something wrong will happen in `selectHost` func finally.

Seems that instead of an empty list, a list like this will returned.
```
[{"Name":"node1","Scores":[],"TotalScore":0},{"Name":"node2","Scores":[],"TotalScore":0}]
```
This is because we initialize NodePluginScores in `Score` for every node 
https://github.com/kubernetes/kubernetes/blob/05f451b58f0d5c54b534acf472fb530fe016ddaf/pkg/scheduler/framework/runtime/framework.go#L979-L983

So seems everything will be fine in `selectHost` since we have nodes in it.

Reply:
Great. Thanks for confirming and adding the test.